### PR TITLE
[MOD-49][Feature]Add moderation setup page and clean up

### PR DESCRIPTION
**Pull request by chrisseto**
screen shot 2017-08-24 at 1 49 41 pm
screen shot 2017-08-24 at 1 49 45 pm
screen shot 2017-08-24 at 1 49 50 pm
screen shot 2017-08-24 at 1 49 57 pm
screen shot 2017-08-24 at 1 50 19 pm

### CODE REVIEW

**Comment by laurenbarker** [comment]
Is there a reason the boxes are super wide?

File: app/styles/app.scss [hunk 1]
```diff
@@ -71,6 +84,7 @@
   .btn-info {
     color: #fff
```
Comment (laurenbarker) [inline app/styles/app.scss]: I think `white` is easier to scan than `#fff`.

File: app/styles/provider/_setup.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
+.moderation-setup {
+  background: #FCFCFC;
```
Comment (laurenbarker) [inline app/styles/provider/_setup.scss]: Could you make these variables so that we don't add 50 slightly different colors in the future? `#FCFCFC` is the default background color that preprints uses.

File: app/styles/provider/_setup.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  padding-bottom: 50px;
+
+  > header {
+    background: #EBEFF0 url('img/reviews-setup-bg.jpg');
+    background-size: cover;
+    padding-bottom: 100px;
+    border-bottom: 1px solid #DDD;
```
Comment (laurenbarker) [inline app/styles/provider/_setup.scss]: Same thing with `#DDD`. It's frequently used for borders between page sections.

File: app/templates/index.hbs [hunk 4]
```diff
@@ -6,6 +6,10 @@
                 {{!TITLE}}
                 <div class="text-center m-t-lg col-xs-12">
                     <div class="p-b-xl reviews-brand"></div>
+                    {{#if model}}
+                        <br>
```
Comment (laurenbarker) [inline app/templates/index.hbs]: I don't think this is an appropriate use of `<br>`; css would be preferred.

File: app/templates/provider/setup.hbs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    <div class="text-bigger alert alert-warning text-center">
+        <i class="fa fa-exclamation-triangle" aria-hidden="true"></i>
+        {{t "setup.once_finalized"}}
+    </div>
+
+    <div class="container">
+        <div class="row m-t-lg">
+            <form class="col-md-offset-2 col-md-8 col-sm-offset-1 col-sm-10">
+                {{#each settingsOptions as |option|}}
```
Comment (laurenbarker) [inline app/templates/provider/setup.hbs]: These would be easier to read if a `max-width` was defined.

File: app/templates/provider/setup.hbs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                            value=choice
+                            attribute=(mut (get this option.name))
+                            }}
+                        {{/each}}
+                    {{/form-block}}
+                {{/each}}
+            </form>
+        </div>
+
+        <br>
```
Comment (laurenbarker) [inline app/templates/provider/setup.hbs]: Also improper use of `<br>`.

File: app/templates/provider/setup.hbs [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    <div class="text-bigger alert alert-warning text-center">
+        <i class="fa fa-exclamation-triangle" aria-hidden="true"></i>
+        {{t "setup.once_finalized"}}
+    </div>
+
+    <div class="container">
+        <div class="row m-t-lg">
+            <form class="col-md-offset-2 col-md-8 col-sm-offset-1 col-sm-10">
+                {{#each settingsOptions as |option|}}
+                    {{#form-block title=(t (concat "setup.settings." option.name ".title")) description=(t (concat "setup.settings." option.name ".description")) disabled=option.disabled}}
```
Comment (laurenbarker) [inline app/templates/provider/setup.hbs]: Could the translation strings be computed in the controller instead of the template? It's a little hard to read but I don't feel strongly either way.

File: app/styles/app.scss [hunk 8]
```diff
@@ -1,8 +1,21 @@
...
+
+// Our 3rd parties
+@import 'osf-style';
+@import 'ember-osf';
+
+// Various other local styles sheets
+@import 'provider';
+
+// Import all styles from pod style components
+@import 'pod-styles';
```
Comment (laurenbarker) [inline app/styles/app.scss]: Woo!

File: app/controllers/provider/setup.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        submit() {
+            REVIEWS_PROVIDER_SETTINGS.forEach(setting => {
+                this.set(`model.${setting.name}`, this.get(setting.name));
+            });
+
+            this.get('model').save().then(() => {
+                return this.transitionToRoute('provider', this.get('model'));
+            }).catch(() => {
+                this.get('toast').error({
+                    msg: `Unable to complete the setup of ${this.get('model.name')}, please try again later.`,
```
Comment (laurenbarker) [inline app/controllers/provider/setup.js]: These should be in the translation file.

File: app/components/form-block/style.scss [hunk 10]
```diff
@@ -0,0 +1,12 @@
+& {
+  margin-top: 20px;
+  background: #FFF;
```
Comment (laurenbarker) [inline app/components/form-block/style.scss]: Same thing here: white > `#FFF`.

File: app/components/form-block/style.scss [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+  padding: 10px 20px 20px;
+  box-shadow: 0 0 3px #ddd;
```
Comment (laurenbarker) [inline app/components/form-block/style.scss]: `#ddd` --> variable

File: app/components/form-block-radiobutton/style.scss [hunk 12]
```diff
@@ -0,0 +1,29 @@
+& {
+  cursor: pointer;
+  padding: 18px;
+  margin: 25px;
+  background: #EEF2F3;
```
Comment (laurenbarker) [inline app/components/form-block-radiobutton/style.scss]: This one should also be a variable. It's used in preprints in a couple of places.

**Review by laurenbarker — COMMENTED** [review]
Looks pretty good, just have some nitpicks. The goal is to have all colors defined as variables -- other than `white` -- so that anyone working on this in the future won't be tempted to add random colors (and it will be obvious if they do).

File: app/components/start-moderating/component.js [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+    setupProvider(provider) {
+        this.set('disabled', true);
+        this.get('action')(provider);
+    },
+
+
+    actions: {
+        submit() {
+            const id = Ember.$('input[type=radio][name=provider]:checked').val();
```
Comment (aaxelb) [inline app/components/start-moderating/component.js]: This makes me sad. Could you make the `form-block-radiobutton` component more generic, usable for these radios too? Would be nice if it could have the provider as value instead of the id, avoiding searching through the array and explaining why `==` instead of `===`

**Comment by chrisseto** [comment]
@laurenbarker wide screen?

screen shot 2017-08-28 at 11 28 20 am

File: app/components/form-block-radiobutton/style.scss (hunk 12 above)
Comment (chrisseto) [reply app/components/form-block-radiobutton/style.scss]: Thoughts on switching to `#EEEEEE`, it's a `$color-light` in osf-style?

File: app/controllers/provider/setup.js [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+        },
+        submit() {
+            REVIEWS_PROVIDER_SETTINGS.forEach(setting => {
+                this.set(`model.${setting.name}`, this.get(setting.name));
+            });
+
+            this.get('model').save().then(() => {
+                return this.transitionToRoute('provider', this.get('model'));
+            }).catch(() => {
+                this.get('toast').error({
```
Comment (aaxelb) [inline app/controllers/provider/setup.js]: Should this also revert the changes to the model, since you took such care not to modify it before submit?

File: app/locales/en/translations.js [hunk 15]
```diff
@@ -17,4 +17,56 @@ export default {
         pre_moderation: `pre-moderation`,
         post_moderation: `post-moderation`,
     },
+    setup: {
+        start: `Start Moderating`,
+        which: `Which provider would you like to set up?`,
+        mulitple_providers: `You're an Admin for Multiple Providers`,
```
Comment (aaxelb) [inline app/locales/en/translations.js]: `s/ulit/ulti/`

**Comment by laurenbarker** [comment]
Still a little wide. The "widely regarded ideal" is 66 characters per line and this is around 86 characters. 66 is contested but pretty much everyone says more than 40cpl less than 70cpl. https://ux.stackexchange.com/questions/108801/what-is-the-best-number-of-paragraph-width-for-readability

File: app/components/form-block-radiobutton/style.scss (hunk 12 above)
Comment (laurenbarker) [reply app/components/form-block-radiobutton/style.scss]: `#EEF2F3` is a little more blue than `#EEEEEE` which looks better with the background color, `#FCFCFC`.
screen shot 2017-08-28 at 11 45 51 am

Comment (chrisseto) [reply app/components/form-block-radiobutton/style.scss]: 👍

File: app/routes/index.js [hunk 16]
```diff
@@ -0,0 +1,13 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+    session: Ember.inject.service(),
+
+    model() {
+        if (!this.get('session.isAuthenticated')) return [];
+        return this.get('store').query('preprint-provider', {
+            'filter[reviews_workflow]': 'None',
```
Comment (aaxelb) [inline app/routes/index.js]: These filters could be written more neatly as an object, though it doesn't much matter:
```
{
  filter: {
    reviews_workflow: 'None', 
    permissions: 'set_up_moderation'
  }
}
```

It's kinda gross that the API expects `'None'` like that... but I guess it works for now.

File: app/styles/app.scss [hunk 17]
```diff
@@ -210,3 +224,8 @@ li.active > a {
 .column-offset {
   height: 120px;
 }
+
+.disabled {
+  pointer-events: none;
```
Comment (aaxelb) [inline app/styles/app.scss]: Thoughts on switching our stylesheets to 4-space indents, for consistency?

File: app/locales/en/translations.js [hunk 18]
```diff
@@ -17,4 +17,56 @@ export default {
         pre_moderation: `pre-moderation`,
         post_moderation: `post-moderation`,
     },
+    setup: {
+        start: `Start Moderating`,
+        which: `Which provider would you like to set up?`,
```
Comment (laurenbarker) [inline app/locales/en/translations.js]: Missed this on the first pass. This should say "Which provider would you like to set up first?" (missing "first").

File: app/components/form-block-radiobutton/component.js [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+    title: null,
+    description: null,
+    value: null,
+    // Namespace for radiobuttons
+    name: null,
+
+    // Bound attribute
+    attribute: null,
+
+    checked: Ember.computed('attribute', function() {
```
Comment (aaxelb) [inline app/components/form-block-radiobutton/component.js]: Should depend on `'value'`, too, in case someone ever tries to use this with mutable value.

File: app/components/start-moderating/component.js [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        this.setupProvider(this.get('providers.firstObject'));
+    },
+
+    setupProvider(provider) {
+        this.set('disabled', true);
+        this.get('action')(provider);
+    },
+
+
```
Comment (aaxelb) [inline app/components/start-moderating/component.js]: extra newline

**Review by aaxelb — COMMENTED** [review]
Looks nice, noticed a few small things and have a request that might be slightly annoying.

File: app/routes/index.js (hunk 16 above)
Comment (chrisseto) [reply app/routes/index.js]: Neato, did not know you could do that

File: app/styles/app.scss (hunk 17 above)
Comment (chrisseto) [reply app/styles/app.scss]: Works for me 👍

File: app/components/start-moderating/component.js (hunk 13 above)
Comment (chrisseto) [reply app/components/start-moderating/component.js]: Honestly, the amount of effort to create a more generic radiobutton component didn't seem worth it :/

Comment (aaxelb) [reply app/components/start-moderating/component.js]: Fair. What about using [ember-radio-button](https://github.com/yapplabs/ember-radio-button)?

**Comment by chrisseto** [comment]
screen shot 2017-08-29 at 9 11 09 am
screen shot 2017-08-29 at 9 11 18 am

**Review by laurenbarker — DISMISSED** [review]
Looks good to me!

File: app/components/start-moderating/template.hbs [hunk 21]
```diff
@@ -0,0 +1,32 @@
+{{t "setup.start"}}
+
+{{#if choiceRequired}}
+
+    {{! Seems to be the best way to style the modal for just this component. Open to new solutions or ideas}}
```
Comment (aaxelb) [inline app/components/start-moderating/template.hbs:5]: Does a component stylesheet not work? Is the modal actually inserted elsewhere?

File: app/locales/en/translations.js [hunk 22]
```diff
@@ -40,5 +40,61 @@ export default {
...
+            title: `Something went wrong`,
+        },
+        settings: {
+            reviewsWorkflow: {
+                title: `Moderation Type`,
+                description: ``,
+                options: {
+                    'pre-moderation': {
+                        title: `Pre-moderation`,
+                        description: `All preprints are placed in a queue for a moderator to accept or reject. Preprints are displayed publicly only after approval.`,
```
Comment (aaxelb) [inline app/locales/en/translations.js]: Should all mentions of preprint use `preprintWord`?

File: app/styles/_util.scss [hunk 23]
```diff
@@ -0,0 +1,4 @@
+.disabled {
+  pointer-events: none;
```
Comment (aaxelb) [inline app/styles/_util.scss:2]: Did you decide no to 4-space indent?

**Review by aaxelb — COMMENTED** [review]
Looks good! Just a couple little things (though `preprintWord` could be annoying, if we go that way).

File: app/locales/en/translations.js (hunk 22 above)
Comment (laurenbarker) [reply app/locales/en/translations.js]: Yes

Comment (aaxelb) [reply app/locales/en/translations.js]: (rhetorical question)

File: app/components/start-moderating/template.hbs (hunk 21 above)
Comment (chrisseto) [reply app/components/start-moderating/template.hbs:5]: It does not, component stylesheets are namespaced but the modal is global.

File: app/services/theme.js [hunk 24]
```diff
@@ -1,42 +1,44 @@
...
         const query = Ember.$.param({
             campaign: `${this.get('id')}-reviews`,
             next: window.location.href
         });
 
         return `${config.OSF.url}register?${query}`;
-    })
+    }),
+
+    onProviderLoad: Ember.observer('provider', function() {
```
Comment (aaxelb) [inline app/services/theme.js:26]: These should be in `ember-osf`'s `theme`, so `ember-preprints` can use it too

File: app/locales/en/translations.js [hunk 25]
```diff
@@ -1,22 +1,23 @@
...
-        brand,
-        brand_name: 'OSF',
-        provider_brand: `{{name}} {{preprintWords.Preprints}}`,
         title: `Title`,
         authors: `Authors`,
         license: 'License',
         pre_moderation: `pre-moderation`,
         post_moderation: `post-moderation`,
     },
+    documentType: {
```
Comment (aaxelb) [inline app/locales/en/translations.js:13]: Seems like this should be in `ember-osf`, replacing its `preprintWords` translations

File: app/locales/en/translations.js [hunk 26]
```diff
@@ -50,5 +51,61 @@ export default {
...
+                    },
+                }
+            },
+            reviewsCommentsAnonymous: {
+                title: `Moderator Comments`,
+                description: `If moderator comments are visible to contributors, the moderator’s name can can be displayed or hidden from the contributors.`,
+                options: {
+                    true: {
+                        title: `Anonymized Comments`,
+                        description: `All comments will be visible to the contributors of the submission, but the moderators name will not be displayed.`,
```
Comment (aaxelb) [inline app/locales/en/translations.js]: `moderator's`

File: app/services/i18n.js [hunk 27]
```diff
@@ -0,0 +1,19 @@
+import Ember from 'ember';
```
Comment (aaxelb) [inline app/services/i18n.js:1]: This file should follow those other bits to ember-osf...

File: app/services/i18n.js [hunk 28]
```diff
@@ -0,0 +1,19 @@
...
+export default I18nService.extend({
+    _globals: {},
+
+    t(key, data = {}) {
+        data = Ember.merge(data, this.get('_globals'));
+        return this._super(key, data);
+    },
+
+    addGlobal(key, value) {
+        this.set(`globals.${key}`, value);
```
Comment (aaxelb) [inline app/services/i18n.js]: `_globals`

File: app/services/i18n.js [hunk 29]
```diff
@@ -0,0 +1,19 @@
+import Ember from 'ember';
+import I18nService from 'ember-i18n/services/i18n';
+
+export default I18nService.extend({
+    _globals: {},
+
+    t(key, data = {}) {
+        data = Ember.merge(data, this.get('_globals'));
```
Comment (aaxelb) [inline app/services/i18n.js]: I think this should be reversed -- user-provided keys should override globals

File: app/router.js [hunk 30]
```diff
@@ -7,7 +7,8 @@ const Router = Ember.Router.extend({
 });
 
 Router.map(function() {
-  this.route('provider', {path: 'preprints/:slug'}, function () {
+  this.route('provider', {path: '/:provider_id'}, function () {
```
Comment (aaxelb) [inline app/router.js]: This still needs to be `preprints/:provider_id`, because reasons @laurenbarker can explain (so we can have a different namespace for non-preprint providers without changing URLs, I think?)

**Review by aaxelb — COMMENTED** [review]
Just a couple things -- mainly the logic that replaces `preprintWords` should go in ember-osf

File: app/services/theme.js (hunk 24 above)
Comment (chrisseto) [reply app/services/theme.js:26]: From Ember OSF:
> Stub so discover page can check for a theme.  Override on your consuming application.

File: app/locales/en/translations.js (hunk 26 above)
Comment (chrisseto) [reply app/locales/en/translations.js]: +1

File: app/services/theme.js (hunk 24 above)
Comment (aaxelb) [reply app/services/theme.js:26]: Right, but since you tore out the `preprintWords` translation stuff from ember-osf, something needs to replace it. It seems reasonable to make `theme` slightly more than a stub. Unless we're planning to duplicate this logic in our ember-preprints branch? Or move the old `preprintWords` logic there?

**Review by aaxelb — DISMISSED** [review]
Looks good 👍 We can figure out the preprintWords stuff later
